backend/core/ai/graph_nodes.py

```
@@ -358,6 +367,76 @@ def detect_anomaly_node(self, state: AgentState) -> Dict[str, Any]:
             print(f"  ✓ 정상 범위: {actual} {operator} {threshold}")
             return {"anomaly_detected": False}
     
+    def _detect_categorical_anomalies(self, state: AgentState) -> Dict[str, Any]:
+        """카테고리형 데이터 이상치 감지 (20개 중 1개도 감지)"""
+        print(f"  🔍 카테고리형 이상치 감지 중...")
```

Ruff (0.13.1)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)